app/models/concerns/orchestration/realm.rb

```diff
@@ -45,26 +45,31 @@ def update_realm
   private
 
   def queue_realm
-    return unless realm? && errors.empty?
+    return unless (realm? || (old && old.realm?)) && errors.empty?
     new_record? ? queue_realm_create : queue_realm_update
   end
 
   def queue_realm_create
-    queue.create(:name => _("Create realm entry for %s") % self, :priority => 50,
+    queue.create(:name => _('Create realm entry for %s') % self, :priority => 50,
                  :action => [self, :set_realm])
   end
 
   def queue_realm_update
-    # Update if the hostgroup is changed or if the realm is changed
-    if self.hostgroup_id_changed? || self.realm_id_changed?
-      queue.create(:name => _("Update realm entry for %s") % self, :priority => 50,
-                   :action => [self, :update_realm])
+    # Remove old entry from realm
+    queue.create(:name => _('Delete realm entry for %s') % self, :priority => 49,
+                 :action => [old, :del_realm]) if old.realm? && realm_id_changed?
```

Comment:
Based on queue_realm and this code, couldn't you delete an existing realm record even when not expected? There's no check if old.realm is the current realm right?

Reply:
I added a test for this. I don't think, this is an issue. `realm_id_changed?` should take care of this. But maybe I'm missing something. Can you please take another look @stbenjam?